### ISSUE TITLE
pngtoy.js:8 Uncaught (in promise) TypeError: Cannot read property 'byteLength' of undefined

### ISSUE THREAD

**Issue by clowerweb** [epistemex/pngtoy#6]
I get the error when opening this image http://i.imgur.com/95TIbv9.png using the following code:

    var png = new PngToy();

    png.fetch("http://i.imgur.com/95TIbv9.png").then(
      function() {png.decode().then(show)}
    );
    
    function show(bmp) {
      var data = bmp.bitmap;
      console.log(data);
    }

**Comment by epistemex** [comment]
Thanks for the report @clowerweb. There is a tEXt chunk before the IEND chunk in the PNG file itself which seem to throw of the parser. The assumption is that no chunks are allowed between the IDAT and IEND chunks except other IDAT chunks, but it seem to be a false assumption as tEXt chunks seem to have no particular order restriction.

As a workaround you can disable strict checking by using the `allowInvalid` option:

    var png = new PngToy({ allowInvalid: true });

In the mean time I will go through the code to reduce the strictness when it comes to textual chunks. I will triple-check the specs and if all OK an update can be expected during the next few days.

**Comment by epistemex** [comment]
There was a tEXt chunk between IDAT and IEND in the PNG file, which is allowed, but threw the parser off. Fixed it to ignore other chunks when compiling IDAT chunks.

**Comment by clowerweb** [comment]
Excellent, great work and thank you!

**Comment by epistemex** [comment]
No problem, and thanks for reporting!

**Comment by clowerweb** [comment]
It works perfectly now! I needed it because I'm building an online IDE tool for [PICO-8](http://www.lexaloffle.com/pico-8.php), which stores game ROMs inside PNG "cartridges", and I was using canvas with context.getImageData, which multiplies the R, G, and B by the alpha, so program code was not decoding properly around the transparent corners of the images. I tried a host of unsuccessful attempts to resolve this for weeks, including using PHP to decode the PNGs on the server, but PHP returns the alpha as a 7-bit (0-127) value, which can't be accurately converted to an 8-bit value. Lastly, I tried placing the image into a WebGL texture buffer and decoding from there, but wasn't able to get that working successfully for my purposes either.

When I was getting the error with this script and having a difficult time finding a similar one that would work, I was going to resort to the most hacky solution ever: reading the image data from canvas, building an array of pixels with 0 alpha, AJAXing the x,y coords over to PHP, then having it build and return a JSON string of RGB values for those pixels (alpha not needed since we'd know they are 0 already). Having this working has saved me a lot of time and effort, so thanks VERY much!
